src/tests/ftest/soak/soak.py

```diff
@@ -0,0 +1,338 @@
+#!/usr/bin/python
+'''
+(C) Copyright 2019 Intel Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+The Government's rights to use, modify, reproduce, release, perform, display,
+or disclose this software are subject to the terms of the Apache License as
+provided in Contract No. B609815.
+Any reproduction of computer software, computer software documentation, or
+portions thereof marked with this legend must also reproduce the markings.
+'''
+
+import os
+import sys
+import json
+import time
+from avocado import Test
+
+sys.path.append('./util')
+sys.path.append('../util')
+sys.path.append('../../../utils/py')
+sys.path.append('./../../utils/py')
+import ServerUtils
+import WriteHostFile
+import IorUtils
+import slurm_utils
+import dmg_utils
+from daos_api import DaosContext, DaosPool, DaosApiError
+
+# global stores intermediate test results
+_soak_results = {}
+
+def job_done(args):
+    """
+    This is a callback function called when a job is done
+
+    handle --which job, i.e. the job ID
+    state  --string indicating job completion status
+    """
+    global _soak_results
+    _soak_results[args["handle"]] = args["state"]
+
+class Soak(Test):
+    """
+    Test class Description: DAOS Soak test cases
+    """
+    def create_pool(self):
+        """
+        Creates a pool that the various tests use for storage.
+        """
+
+        createmode = self.params.get("mode", '/run/pool1/createmode/*/')
+        createuid = os.geteuid()
+        creategid = os.getegid()
+        createsetid = self.params.get("setname", '/run/pool1/createset/')
+        createsize = self.params.get("size", '/run/pool1/createsize/')
+        self.createsvc = self.params.get("svcn", '/run/pool1/createsvc/')
+
+        self.pool = DaosPool(self.context)
+        self.pool.create(createmode, createuid, creategid,
+                         createsize, createsetid, None, None,
+                         self.createsvc)
+
+    def get_ior_job(self, spec, processes, name):
```

Comment:
Shouldn't we have a generic class for different "job" and then implement it for IOR, mdtest, ...

Reply:
I think this is a case of poor naming.  The function builds the command line string used to run ior from the test parameters.  There is not really the member data that would be the basis for a class.  But I expect we will have  other functions to build other command strings.  Renamed the function to build_ior_cmd().  There is a separate module to do most of this and this function just handles getting the parameters from the yaml file.